pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -61,9 +61,14 @@ func (r *Registration) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeCla
 		}
 		return reconcile.Result{}, fmt.Errorf("getting node for nodeclaim, %w", err)
 	}
+	if _, hasStartupTaint := lo.Find(node.Spec.Taints, func(t v1.Taint) bool {
+		return t.MatchTaint(&v1beta1.UnregisteredNoExecuteTaint)
+	}); !hasStartupTaint {
+		return reconcile.Result{}, fmt.Errorf("missing required startup taint, %s", v1beta1.UnregisteredTaintKey)
+	}
 	ctx = log.IntoContext(ctx, log.FromContext(ctx).WithValues("Node", klog.KRef("", node.Name)))
 	if err = r.syncNode(ctx, nodeClaim, node); err != nil {
-		return reconcile.Result{}, fmt.Errorf("syncing node, %w", err)
+		return reconcile.Result{Requeue: true}, err
```

Comment:
We should automatically requeue with backoff if the error is not true, is there a reason we're setting `Requeue` explicitly here?

Reply:
In this case we're returning the error though so don't we have to set requeue?

Reply:
Yep, if the error is non-nil the result is ignored and it enters expoenential backoff: [ref](https://github.com/kubernetes-sigs/controller-runtime/blob/12cc8d59fabe3a1dbbd927ae6aea077cf7962267/pkg/reconcile/reconcile.go#L95-L97)